backend/lamb/assistant_sharing_router.py

```diff
@@ -121,34 +121,51 @@ def sync_assistant_to_owi_group(assistant_id: int, db_manager: LambDatabaseManag
         if user:
             user_emails.append(user['user_email'])
     
-    # Create or update OWI group
-    group_name = f"assistant_{assistant_id}_shared"
+    # Use the ORIGINAL assistant group (assistant_X, not assistant_X_shared)
+    group_name = f"assistant_{assistant_id}"
     
-    # Check if group exists
-    existing_groups = group_manager.get_all_groups()
-    group_id = None
-    
-    for group in existing_groups:
-        if group.get('name') == group_name:
-            group_id = group['id']
-            break
+    # Check if group exists using get_group_by_name (more efficient and avoids duplicates)
+    existing_group = group_manager.get_group_by_name(group_name)
+    group_id = existing_group['id'] if existing_group else None
     
     if not group_id:
-        # Create new group
-        owner_manager = OwiUserManager()
-        owner_user = owner_manager.get_user_by_email(assistant.owner)
-        if owner_user:
-            result = group_manager.create_group(
-                name=group_name,
-                description=f"Shared access for assistant {assistant.name}",
-                user_id=owner_user['id']
-            )
-            group_id = result.get('id') if result.get('status') == 'success' else None
+        # Before creating, check if there are any duplicate groups with this name
```

Comment:
Simplify. We should assume that there are no duplicates. We will run the `migrate_assistant_groups.py` first.